content/blog/dynamic-method.md

```diff
@@ -0,0 +1,99 @@
++++
+title = "Dynamic method in ruby"
+slug = "dynamic-method"
+date = 2023-10-27T13:31:36+05:30
+image = "/images/2023/10/methodmissing.jpg"
+draft = false
+authors = ["Nitin Rajkumar"]
+description = "Dynamic and ghost methods in ruby"
+tags = ["Software Craftsmanship", "GraphQL"]
+categories = ["Software Craftsmanship", "Ruby", "Dynamic Dispatch", "Ghost Methods"]
+type = ""
++++
+
+In Ruby, the dynamic nature of method dispatch is one of its key features. Method calls are resolved at runtime based on the actual type of the object. This flexibility allows for polymorphic and dynamic behavior
+#### Dynamic Dispatch
+```ruby
+	class Number
+		def square(argument)
+			argument ** 2
+		end
+		def cube(argument)
+			argument ** 3
+		end
+	end
+	number = Number.new
+	number.square(3) #=> 9
+```
+
+Method can be called using dot notation but this restricts making the method call being dynamic. 
+There is another way to call a method that is using `send`. It accepts method name and it's arguments as arguments.
```

Comment:
instead of "it's arguments as arguments"

"the method's arguments as it own arguments"